test/unit/assorted/client.test.js

```diff
@@ -38,7 +38,10 @@ describe('Client (unit)', function () {
 
     return client.connect().then(() => {
       expect(handshake).to.have.nested.property('client.driver');
-      expect(handshake).nested.property('client.driver.name').to.equal('nodejs|mongoose');
+      expect(handshake)
+        .nested.property('client.driver.name')
+        // TODO(NODE-4979): Our tests depend on mongodb-legacy
```

Comment:
I think this will be the case even after they don't depend, since it'll be a supported use case, same as mongoose really

Reply:
Updated the comment to be clearer, looking back on that months from now the context was missing

Reply:
I think you misunderstood my comment - even after we complete NODE-4979, we want to continue using our library to test the wrapper (in a separate build, see NODE-4980); only once the legacy wrapper is no longer supported and we don't test it in the CI (which we don't have a timeline or a ticket for at the moment, but I suppose we could create a placeholder ticket like we have for legacy hello that gets bumped from major version to major version) can we get rid of that expectation.

Reply:
fixed the test to handle both cases so it shouldn't fail when we flip it